parsers/verilog.c

```diff
@@ -1306,11 +1316,13 @@ static void findTag (tokenInfo *const token)
 {
 	verbose ("Checking token %s of kind %d\n", vStringValue (token->name), token->kind);
 
-	if (currentContext->kind != K_UNDEFINED)
+	if (currentContext->kind != K_UNDEFINED && (token->kind == K_END || token->kind == K_END_DE))
 	{
 		/* Drop context, but only if an end token is found */
 		dropEndContext (token);
 	}
+	if (token->kind == K_END_DE)
+		return;
```

Comment:
My understanding is that this line is called only for broken (invalid syntax) input.
In that case, putting verbose() here may help us in the future.

Reply:
This line is called for correct input, endfunction, endmodule, endtask, ..., etc.

***

Your comment reminds me one more question.
I think we should call notice() instead of verbose() for broken input.
And every parser should use same format something like this [this](https://www.gnu.org/prep/standards/standards.html#Errors).

How do you think?

Reply:
I think notice() is too noisy. Furthermore, verbose is also noise for the purpose. Remember ctags will deal with large codebases.



Printing such information has two purposes.

* reducing pointless bug reports
  We know the motto, "if the input is broken, ctags may emit garbages". However, users may not know.
  As the result, we have chances to receive bug reports of ctags to that broken input is given.
  The notice can notify that broken input is given to ctags.

* debugging
  This is for us, not for users. For this purpose, verbose() and notice() are not suitable.
   We can use macros in trace.h. See parsers/fypp.c.
   The macros can be used only with ctags built with ./configure --enable-debugging.
    --_trace=<LANG> must be passed to ctags at runtime.